### ISSUE TITLE
Module declarations missing from generated app

### ISSUE THREAD

**Issue by habiblawal1** [OpenLiberty/start.openliberty.io#262]
**Describe the bug**  
Eclipse was unable to resolve a lot of jakarta classes such as `jakarta.ws.rs.RestApplication` when I imported the generated gradle project.

**Steps to Reproduce**  
Generate a starter project with all the default settings, and then import it into Eclipse as an **Existing Gradle Project** with Java 17 as the JDK.

**Expected behavior**  
The project should show up with no errors and an Eclipse gradle build should run successfully.

**Diagnostic information:**
On examination the error pointed to a module visibility problem. 
I was able to resolve it by using the following `module-info.java'

```
module app-name {
	requires jakarta.ws.rs;
}
```

**Comment by natalie-bernhard** [comment]
Hi, unfortunately there is not a large use case for JPMS for our users, so we do not support it directly out of the box when using the starter.

**Comment by scottkurz** [comment]
@habiblawal1  thank you for the question.

Let me just note that we do offer Eclipse IDE support for working with Jakarta & MicroProfile Gradle application projects in Liberty via our [Liberty Tools](https://marketplace.eclipse.org/content/liberty-tools) feature.   I know one can certainly import a Open Liberty starter Gradle app into Eclipse with the default, "out-of-the-box" Gradle/buildship config and it will build fine.

I'm not familiar enough with Gradle + JPMS to know if there's anything else to point out here, but let me just offer that you can feel free to contact us on our Gitter channel if you want to discuss more: 
https://app.gitter.im/#/room/#OpenLiberty_developer-experience:gitter.im

**Comment by tjwatson** [comment]
What version of Eclipse are you using?

I tried reproducing this with 2024-03 release for Java developers with no luck:  https://www.eclipse.org/downloads/packages/release/2024-03/r/eclipse-ide-java-developers

> Eclipse was unable to resolve a lot of jakarta classes such as jakarta.ws.rs.RestApplication

I don't think `jakarta.ws.rs.RestApplication` exists in EE 10.  Did you mean `jakarta.ws.rs.core.Application`?  When I import the gradle project created by the Liberty starting using Java 17 or Java 21 I see no `module-info.java` source file created and Eclipse resolves the project classpath to all the APIs from maven central that are included with the `jakarta.platform:jakarta.jakartaee-api:10.0.0` dependency.

On the other hand if I create a `module-info.java` source file with an empty body like this:

```
module appName {
}
```
Then I start to see compile errors in my workspace when trying to resolve `jakarta.ws.rs.core.Application`.  But to get that to work correctly I had to set the gradle classpath container (in the `.classpath` of the project) to have the `module` attribute:

```
	<classpathentry kind="con" path="org.eclipse.buildship.core.gradleclasspathcontainer">
		<attributes>
			<attribute name="org.eclipse.jst.component.nondependency" value=""/>
			<attribute name="module" value="true"/>
		</attributes>
	</classpathentry>
```
This can be done with the project settings for the projects Java Build Path.


This is expected because the Java compiler starts enforcing JPMS class path rules as soon as you have a `module-info` in your class path for the project and you have informed Eclipse to treat the classpath entry as a module classpath.  If your project is setup to have a Module path in Eclipse then Eclipse will require you to have a `module-info` and also require it to specify what it `requires` in order to compile.
